src/main/java/org/dependencytrack/metrics/Metrics.java

```diff
@@ -36,11 +46,30 @@
  */
 public final class Metrics {
 
+    protected static QueryManager qm;
+
     private Metrics() {
     }
 
     public static double inheritedRiskScore(final int critical, final int high, final int medium, final int low, final int unassigned) {
-        return (double) ((critical * 10) + (high * 5) + (medium * 3) + (low * 1) + (unassigned * 5));
+        try (final var qm = new QueryManager()) {
+            ConfigProperty risk_score_critical_property = qm.getConfigProperty(CUSTOM_RISK_SCORE_CRITICAL.getGroupName(), CUSTOM_RISK_SCORE_CRITICAL.getPropertyName());
+            ConfigProperty risk_score_high_property = qm.getConfigProperty(CUSTOM_RISK_SCORE_HIGH.getGroupName(), CUSTOM_RISK_SCORE_HIGH.getPropertyName());
+            ConfigProperty risk_score_medium_property = qm.getConfigProperty(CUSTOM_RISK_SCORE_MEDIUM.getGroupName(), CUSTOM_RISK_SCORE_MEDIUM.getPropertyName());
+            ConfigProperty risk_score_low_property = qm.getConfigProperty(CUSTOM_RISK_SCORE_LOW.getGroupName(), CUSTOM_RISK_SCORE_LOW.getPropertyName());
+            ConfigProperty risk_score_unassigned_property = qm.getConfigProperty(CUSTOM_RISK_SCORE_UNASSIGNED.getGroupName(), CUSTOM_RISK_SCORE_UNASSIGNED.getPropertyName());
+
+            int risk_score_critical_val = Integer.valueOf(risk_score_critical_property.getPropertyValue());
+            int risk_score_high_val = Integer.valueOf(risk_score_high_property.getPropertyValue());
+            int risk_score_medium_val = Integer.valueOf(risk_score_medium_property.getPropertyValue());
+            int risk_score_low_val = Integer.valueOf(risk_score_low_property.getPropertyValue());
+            int risk_score_unassigned_val = Integer.valueOf(risk_score_unassigned_property.getPropertyValue());
+
+            return (double) ((critical * risk_score_critical_val) + (high * risk_score_high_val) + (medium * risk_score_medium_val) + (low * risk_score_low_val) + (unassigned * risk_score_unassigned_val));
+
+        } catch (Exception e) {
+            return (double) ((critical * 10) + (high * 5) + (medium * 3) + (low * 1) + (unassigned * 5)); 
+        }
```

Comment:
Note, I think this method isn't used anymore.

We had to move metrics calculation to stored procedures on the database, since moving all this data between database and API server was too resource-intensive.

This corresponding SQL for this method now lives here: https://github.com/DependencyTrack/hyades-apiserver/blob/df57c53cd2bb91f6bc8e642d6a1bea353153da9e/src/main/resources/migration/procedures/function_calc-risk-score.sql#L1-L19

It is invoked during component metrics calculation, here:

https://github.com/DependencyTrack/hyades-apiserver/blob/df57c53cd2bb91f6bc8e642d6a1bea353153da9e/src/main/resources/migration/procedures/procedure_update-component-metrics.sql#L123

In theory, you can get your desired outcome with something like this (not tested):

```sql
WITH "CUSTOM_SCORES" AS (
  SELECT "PROPERTYVALUE"::INT AS "value"
       , "PROPERTYNAME" AS "name"
    FROM "CONFIGPROPERTY"
   WHERE "PROPERTYGROUP" = 'risk-score'
     AND "PROPERTYTYPE" = 'INTEGER'
)
SELECT (
  ("critical" * (SELECT "value" FROM "CUSTOM_SCORES" WHERE "name" = 'riskscore.critical'))
  -- TODO: Repeat for the other scores
  + ("high" * 5) 
  + ("medium" * 3)
  + ("low" * 1)
  + ("unassigned" * 5)
)::NUMERIC;
```

This uses a [CTE](https://www.postgresql.org/docs/current/queries-with.html) to get the relevant config values.

Reply:
You can still test this in mostly-Java. The [metrics task tests](https://github.com/DependencyTrack/hyades-apiserver/blob/main/src/test/java/org/dependencytrack/tasks/metrics/ComponentMetricsUpdateTaskTest.java) all call the stored procedures behind the scenes. So you can do the test setup, and assertions of your desired outcome as usual.

Reply:
i removed the qm line and now the metricstests pass, thanks. though maybe i should revert it all since that function isn't used anymore. 

i added the sql, but i didn't know if that quotes are still needed, and if it should be in the create function statement or a separate one. is there a way to test just the sql?

Reply:
The SQL in your most recent commit looks like it should work. There are test failures now because the new `ConfigProperty`s are not created in those tests. Once you create the properties for them, it should work!

> is there a way to test just the sql?

You can always create / replace the SQL function in a database and manually invoke it. But IMO the easiest way to test it is to use JUnit, as done in the metrics task tests I linked above. The functions are automatically deployed when the test start.

Reply:
moved the logic to sql and updated ComponentMetricsUpdateTaskTest